### PR TITLE
feat(key-management-service): add kms dashboard

### PR DESCRIPTION
**Pull request by vovh**
ref: MANAGER-13567

| Question         | Answer
| ---------------- | ---
| Branch?          | feat/key-management-service
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-13567
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

As a customer, I want to view KMS details, and make actions on it (edit information and encryption methods)

### CODE REVIEW

File: packages/manager/apps/key-management-service/src/api/services/useUpdateOkmsName.ts
Comment (darsene) [inline packages/manager/apps/key-management-service/src/api/services/useUpdateOkmsName.ts:1]: to move in `hooks` folder

File: packages/manager/apps/key-management-service/src/shell.ts
Comment (darsene) [inline packages/manager/apps/key-management-service/src/shell.ts:1]: useless file, already defined in `@ovh-ux/manager-react-shell-client`, you can remove it

**Review by darsene — APPROVED** [review]

File: packages/manager/apps/key-management-service/src/components/Modal/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
+import { OsdsModal } from '@ovhcloud/ods-components/react';
+import React, { useRef } from 'react';
+
+const Modal: React.FC<React.PropsWithChildren & {
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager/apps/key-management-service/src/components/Modal/Modal.tsx:5]: already in progress in another PR 👍

File: packages/manager/apps/key-management-service/src/components/Listing/ListingCells.tsx [hunk 2]
```diff
@@ -3,11 +3,26 @@ import {
...
+import { ODS_TEXT_COLOR_INTENT } from '@ovhcloud/ods-components';
+import { useNavigate } from 'react-router-dom';
 import { useTranslation } from 'react-i18next';
 import { OKMS } from '@/interface';
 
 export const DatagridCellName = (props: OKMS) => {
-  return <DataGridTextCell>{props.iam.displayName}</DataGridTextCell>;
+  const navigate = useNavigate();
+  return (
+    <div>
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager/apps/key-management-service/src/components/Listing/ListingCells.tsx:15]: we have to find why we need to wrap with a div ? Nicolas has the same issue

File: packages/manager/apps/key-management-service/src/api/hooks/useUpdateOkmsName.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      const { data: servicesId } = await queryClient.fetchQuery<
+        ApiResponse<number[]>
+      >({
+        queryKey: getOkmsServiceIdQueryKey({ okms }),
+        queryFn: () => getOkmsServiceId({ okms }),
+      });
+      return updateOkmsName({ serviceId: servicesId[0], displayName });
+    },
+    onSuccess: () => {
+      setTimeout(
```
Comment (aboungnaseng-ovhcloud) [inline packages/manager/apps/key-management-service/src/api/hooks/useUpdateOkmsName.ts:44]: why do you need a setTimeOut ?

Comment (darsene) [reply packages/manager/apps/key-management-service/src/api/hooks/useUpdateOkmsName.ts]: @aboungnaseng-ovhcloud it's fixed

File: packages/manager/apps/key-management-service/src/components/Modal/EditNameModal.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+import Modal from './Modal';
+import { OKMS } from '@/interface';
+
+interface DeleteModalProps {
+  okms: OKMS;
+  toggleModal: (showModal: boolean) => void;
+  onEditName: (okms: OKMS) => void;
+}
+
+const EditNameModal = ({ okms, toggleModal, onEditName }: DeleteModalProps) => {
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager/apps/key-management-service/src/components/Modal/EditNameModal.tsx:29]: It will be rework with a generic component 👍

**Review by aboungnaseng-ovhcloud — APPROVED** [review]

**Pull request merged**